frontend/src/app/person-notes/person-notes.component.spec.ts

```diff
@@ -1,7 +1,294 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { async, fakeAsync, TestBed, tick } from '@angular/core/testing';
 
 import { PersonNotesComponent } from './person-notes.component';
+import { FormsModule } from '@angular/forms';
+import { HttpClientModule } from '@angular/common/http';
+import { NoteComponent } from '../note/note.component';
+import { PersonNoteService } from '../person-note.service';
+import { UserService } from '../user.service';
+import { JwtInterceptorService } from '../jwt-interceptor.service';
+import { PersonModel } from '../models/person.model';
+import { NoteModel } from '../models/note.model';
+import { Observable } from 'rxjs/Observable';
+import { ConfirmService } from '../confirm.service';
+import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
+import 'rxjs/add/operator/debounceTime';
+import { Subject } from 'rxjs/Subject';
+import { By } from '@angular/platform-browser';
+import createSpyObj = jasmine.createSpyObj;
+import { UserModel } from '../models/user.model';
 
 describe('PersonNotesComponent', () => {
-  // TODO
+
+  const person = { id: 42 } as PersonModel;
+  let notes: Array<NoteModel>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [FormsModule, HttpClientModule, NgbModule.forRoot()],
+      declarations: [PersonNotesComponent, NoteComponent],
+      providers: [
+        // For a reason that I can't understand, using the real implementation and spying on its update
+        // method doesn't work. The real method is being called.
+        { provide: PersonNoteService, useValue: createSpyObj('PersonService', ['list', 'create', 'update', 'delete']) },
```

Comment:
Actually, they're not, even now. I realized, looking at the code coverage, that saveNote() is never being called. The closure passed to whenStable().then() is never executed.